### ISSUE TITLE
M1 architecture support for node-based images

### ISSUE THREAD

**Issue by thclark** [pre-commit/pre-commit#1786]
Might be worth mentioning in brief [here](https://pre-commit.com/#node) that node based hooks won't run on M1 macs yet.

This is an upstream issue with nodenv, not precommit - I've filed an issue with the stacktrace [here](https://github.com/ekalinin/nodeenv/issues/280) and it doesn't look like a bad fix so should get ironed out soon.

But in the meantime, hooks like `mirrors-eslint` won't work.

**Comment by asottile** [comment]
As a workaround if you have `node` installed at the system it should automatically use `-n system`

**Comment by thclark** [comment]
Brilliant, thanks @asottile!
Node is natively installable on M1 via brew, for anyone looking.

**Comment by boertel** [comment]
For whoever ends up on this issue, I had the same issue and I couldn't figure out how to use `-n system`. I found out that adding: `language_version: system` to my eslint hook configuration works. 

Here is my `.pre-commit-config.yaml`:
```
repos:
  - repo: https://github.com/pre-commit/mirrors-eslint
    rev: v7.19.0
    hooks:
    -   id: eslint
        language_version: system
        additional_dependencies:
        -   eslint@6.8.0
        -   eslint-loader@3.0.3
        -   eslint-plugin-react@7.19.0
        -   eslint-plugin-import@2.20.1
        -   eslint-plugin-flowtype@4.6.0
        -   eslint-plugin-jsx-a11y@6.2.3
        -   eslint-plugin-react-hooks@1.6.1
        -   babel-eslint@10.1.0
```

**Comment by asottile** [comment]
`language_version: system` should be automatic if you're using a new-enough pre-commit -- I suspect you're on an old version?

**Comment by thclark** [comment]
Or have a default language version specified. Either way, that's helpful, thanks @boertel - helpful!

**Comment by boertel** [comment]
@asottile I'm using pre-commit 2.10.1

**Comment by asottile** [comment]
yeah that'd be the issue -- shims are intentionally avoided because they break in mysterious ways

**Comment by thclark** [comment]
This is now fixed in the upstream [1.6.0 version of nodeenv](https://github.com/ekalinin/nodeenv/releases/tag/1.6.0)
